File: men_clothes-backend/shopapp/src/main/java/com/project/shopapp/responses/DetailProductResponse.java
package com.project.shopapp.responses;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.project.shopapp.models.*;
import lombok.*;

import java.util.ArrayList;
import java.util.List;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class DetailProductResponse {
    private Long id;
    private String name;
    private Float price;
    private String thumbnail;
    private String description;
    @JsonProperty("category")
    private Category category;
    @JsonProperty("color")
    private Color color;
    @JsonProperty("size")
    private Size size;
}

File: men_clothes-angular/src/app/components/responses/product/detail.product.response.ts
export interface Category {
  id: number;
  name: string;
}

export interface Color {
  id: number;
  name: string;
}

export interface Size {
  id: number;
  name: string;
}

export interface DetailProductResponse {
  id: number;
  name: string;
  price: number;
  thumbnail: string;
  description?: string | null; // Giá trị có thể là null
  category: Category;
  color: Color;
  size: Size;
}
